File: src/leetcode/P0278.java
package leetcode;

public class P0278 {
    public static void main(String[] args) {
        int a = 1595065044;
        a -= a/2;
        System.out.println(a);
        System.out.println(firstBadVersion(2126753390));
    }
    public static int firstBadVersion(int n) {
        if (n == 1) {
            return 1;
        }
        int res = 0;
        int next = n/2 + 1;



        return res;
    }

    public static boolean isBadVersion(int n) {
        return n >= 1702766719;
    }
}

File: ruby/leetcode/problems/P0278.rb
# 278. First Bad Version
def is_bad_version(version)
  bad = 1
  bad <= version
end

def first_bad_version(n)
  left = 0
  right = n
  mid = -1
  while left <= right
    mid = (left + right) / 2
    res = is_bad_version(mid)
    if !is_bad_version(mid - 1) && res
      break
    end
    if res
      right = mid - 1
    else
      left = mid + 1
    end
  end

  mid
end

puts first_bad_version(1)